pkg/resource/deploy/deploytest/resourcemonitor.go

```diff
@@ -235,8 +236,12 @@ func (rm *ResourceMonitor) RegisterResource(t tokens.Type, name string, custom b
 	}
 
 	ctx := context.Background()
-	if len(opts.GrpcRequestHeaders) > 0 {
-		ctx = metadata.NewOutgoingContext(ctx, metadata.New(opts.GrpcRequestHeaders))
+	if len(opts.GrpcRequestHeaders) > 0 || opts.RegisteredAt != "" {
+		md := metadata.New(opts.GrpcRequestHeaders)
+		if opts.RegisteredAt != "" {
+			md.Set("source-position", opts.RegisteredAt)
```

Comment:
Kinda feel it would be better to keep the metadata name and the state name in sync. Why "RegisteredAt" instead of just "SourcePosition"?

Reply:
> Why "RegisteredAt" instead of just "SourcePosition"?

The former felt a little bit more descriptive than the latter. Happy to keep `SourcePosition`, though.

Reply:
I like `SourcePosition` - espevilly for the state file, where the notion of “registered” isn’t really a user facing concept.

Reply:
Done.